### PR TITLE
protocol: create separate request context for fetch job

### LINKED ISSUE #2182: RequestHttpJob If do not unintercept will infinite loop
If do not unintercept will infinite loop 
Unable to intercept http unintercept after

```javascript
var protocol = require('protocol');
   protocol.interceptProtocol('http', function(request) {
     var url = request.url;
    //  protocol.uninterceptProtocol("http")
     return new protocol.RequestHttpJob({url:url+"?dd=1"});
   });
```

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #2182

### CODE REVIEW

**Comment by deepak1556** [comment]
@zcbenz have made the changes.

File: atom/browser/api/atom_api_protocol.cc [hunk 1]
```diff
@@ -126,13 +144,15 @@ class CustomProtocolRequestJob : public AdapterRequestJob {
       } else if (name == "RequestHttpJob") {
         GURL url;
         std::string method, referrer;
+        net::URLRequestContextGetter* getter;
```
Comment (zcbenz) [inline atom/browser/api/atom_api_protocol.cc]: It should be initialized to nullptr.

**Comment by zcbenz** [comment]
When user didn't specify a `session` in `RequestHttpJob`, current request context should be used.

File: atom/browser/api/atom_api_protocol.cc (hunk 1 above)
Comment (deepak1556) [inline atom/browser/api/atom_api_protocol.cc]: have handled the cases in the `Converter<>`

**Comment by deepak1556** [comment]
Isnt using the current request context the cause for infinite loop, the interceptor and the request from fetch job will be in same context ? I imagined when session is null or undefined create a new context or else use from the session provided. Did i get that wrong ?

**Comment by zcbenz** [comment]
If user is starting a `http` request while intercepting the `http` protocol, it is expected that it causes an infinite loop. It is like wring a [fork bomb](https://en.wikipedia.org/wiki/Fork_bomb), it locks the system but it is the expected behavior.

**Comment by deepak1556** [comment]
Fixed, Thanks!

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**